sdk/azidentity/bearer_token_policy_test.go

```diff
@@ -174,15 +176,22 @@ func TestBearerPolicy_GetTokenFailsNoDeadlock(t *testing.T) {
 		HTTPClient:    srv,
 		AuthorityHost: srv.URL(),
 		Retry: azcore.RetryOptions{
-			// leaving TryTimeout at zero will trigger a deadline exceeded error causing GetToken() to fail
-			RetryDelay:    50 * time.Millisecond,
-			MaxRetryDelay: 500 * time.Millisecond,
-			MaxRetries:    3,
+			// leaving TryTimeout at as a very small value will trigger a deadline exceeded error causing GetToken() to fail
+			TryTimeout: 1 * time.Nanosecond,
+			MaxRetries: -1,
```

Comment:
This will trigger no retries which invalidates the test.

Reply:
Yes I noticed this. I'll reverse it, but I'm looking into it and I've made sure I have all of the same configurations as before and this test is still failing. Could something have changed down the pipeline? (I'm also thinking of skipping this test and opening an issue to resolve this in it's own PR)

Reply:
I think we need to understand what's happening with the failure.  The change to default values should have had zero impact (i.e. the old/new pattern should yield identical default values).

Reply:
The problem is that now, instead of getting an error the test doesnt receive an error and gets the successful response.

Reply:
Set TryTimeout as a negative value like you recommended. Done.